api/envoy/extensions/filters/network/http_connection_manager/v3/http_connection_manager.proto

```diff
@@ -506,6 +507,38 @@ message HttpConnectionManager {
   //
   // 3. Tracing decision (sampled, forced, etc) is set in 14th byte of the UUID.
   RequestIDExtension request_id_extension = 36;
+
+  // Configuration of local reply returned by Envoy. Allows to specify mappings and format of
+  // response.
+  LocalReplyConfig local_reply_config = 38;
+}
+
+message LocalReplyConfig {
+  // Configuration of list of mappers which allows to filter and change local response.
+  // The client will iterate through mappers until first match.
+  repeated ResponseMapper mapper = 1;
+
+  // Allows to define custom format of local reply.
+  // It is allowed to use :ref:`command operators <config_access_log_command_operators>`
+  // which will be resolved to actual data.
+  type.v3.StringOrJson format = 2;
+}
+
+message ResponseMapper {
```

Comment:
I could potentially see wanting format to set on a per-mapper basis. Should we move `format` here? This would allow us to just have a catch-all mapper? This might also yield too much duplication. WDYT?

Reply:
Yes I think that makes sense, thanks.